modmesh/pilot/airfoil/__init__.py

```diff
@@ -0,0 +1,264 @@
+        line is calculated (:math:`m` is the maximum camber, :math:`p` is the
+        position of the maximum camber):
+
+        .. math:
+
+          y_c = \frac{m}{p^2}(2px - x^2), \quad 0 \le x \le p \\
+          y_c = \frac{m}{(1-p)^2}(1 - 2p + 2px - x^2), \quad p \le x \le 1
+
+        The thickness needs to be perpendicular to the camber line, the upper
+        and lower profile should be calculated by:
+
+        .. math:
+
+          x_u = x - y_t\sin\theta , \; y_u = y_c + y_t\cos\theta \\
+          x_l = x + y_t\sin\theta , \; y_l = y_c - y_t\cos\theta
+
+        The angle :math:`\theta` should be determined by:
+
+        .. math:
+
+          \theta = \arctan\frac{\mathrm{d}y_c}{\mathrm{d}x} \\
+          \frac{\mathrm{d}y_c}{\mathrm{d}x} = \frac{2m}{p^2}(p-x),
+            \quad 0 \le x \le p \\
+          \frac{\mathrm{d}y_c}{\mathrm{d}x} = \frac{2m}{(1-p)^2}(p-x),
+            \quad p \le x \le 1
+
+        :param xc: x coordinate at center/camber
+        :return: xu, yu, xl, yl
+        """
+        yt = self.calc_yt(xc)
+        pos = self.pos
+        if pos == 0:
+            return xc, yt, xc, -yt
+        else:
+            camber = self.camber
+            if xc <= pos:
+                xc1 = xc
+                yc1 = camber / np.power(pos, 2) * xc1 * (2 * pos - xc1)
+                dyc1_dx = camber / np.power(pos, 2) * (2 * pos - 2 * xc1)
+                yc = yc1
+                dyc_dx = dyc1_dx
+            else:
+                xc2 = xc
+                yc2 = camber / np.power(1 - pos, 2) * (1 - 2 * pos + xc2) * (
+                        1 - xc2)
+                dyc2_dx = camber / np.power(1 - pos, 2) * (2 * pos - 2 * xc2)
+                yc = yc2
+                dyc_dx = dyc2_dx
+
+            theta = np.arctan(dyc_dx)
+
+            return (xc - yt * np.sin(theta), yc + yt * np.cos(theta),
+                    xc + yt * np.sin(theta), yc - yt * np.cos(theta))
+
+    def calc_points(self, npoint):
+        """
+        :param npoint: Number of sample points on the center/camber
+        :return: Arrays for xu, yu, xl, yl
+        """
+        # Make xc array.
+        if self.cosine_spacing:
+            _ = np.linspace(0.0, np.pi, npoint + 1, dtype='float64')
+            xcarr = 0.5 * (1.0 - np.cos(_))
+        else:
+            xcarr = np.linspace(0.0, 1.0, npoint + 1, dtype='float64')
+        # Prepare and populate arrays for (xu, yu), (xl, yl)
+        xuarr = np.empty_like(xcarr)
+        yuarr = np.empty_like(xcarr)
+        xlarr = np.empty_like(xcarr)
+        ylarr = np.empty_like(xcarr)
+        for i, xc in enumerate(xcarr):
+            xuarr[i], yuarr[i], xlarr[i], ylarr[i] = self.calc_ul(xc)
+        # Make return array.
+        xrarr = np.concatenate([xuarr[::-1], xlarr[1:]])
+        yrarr = np.concatenate([yuarr[::-1], ylarr[1:]])
+        ret = np.vstack([xrarr, yrarr]).T.copy()
+        return ret
+
+
+class Naca4Sampler(object):
```

Comment:
`Naca4Sampler` provides `draw_line()` and `draw_cbc()` and uses "world" to draw.  This will need more work in the future for interaction with GUI.